src/main/java/frc/trigon/robot/subsystems/turret/Turret.java

```diff
@@ -0,0 +1,50 @@
+package frc.trigon.robot.subsystems.turret;
+
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+
+public class Turret extends SubsystemBase {
+    private final static Turret INSTANCE = new Turret();
+    private final TurretIO turretIO = TurretIO.generateIO();
+    private final TurretInputsAutoLogged turretInputs = new TurretInputsAutoLogged();
+
+    public static Turret getInstance() {
+        return INSTANCE;
+    }
+
+    private Turret() {
+    }
+
+    @Override
+    public void periodic() {
+        turretIO.updateInputs(turretInputs);
+        Logger.processInputs("Turret", turretInputs);
+    }
+
+    /**
+     * @param robotPosition position of the robot on the field
+     * @return a command that sets the turret to look at the centre hub
+     */
+    void setMotorPowerFromPosition(Pose2d robotPosition) {
```

Comment:
give this a better name. Every command at the end just sets motor power, but why does this one do that? What's the goal? It's like calling the command that sets the target angle position of the arm set target power command.